// @ionic-native/firebase-analytics/__ivy_ngcc__/ngx/index.js
import { __decorate, __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var FirebaseAnalytics = /** @class */ (function (_super) {
    __extends(FirebaseAnalytics, _super);
    function FirebaseAnalytics() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FirebaseAnalytics.prototype.logEvent = function (name, params) { return cordova(this, "logEvent", { "sync": true }, arguments); };
    FirebaseAnalytics.prototype.setUserId = function (id) { return cordova(this, "setUserId", { "sync": true }, arguments); };
    FirebaseAnalytics.prototype.setUserProperty = function (name, value) { return cordova(this, "setUserProperty", { "sync": true }, arguments); };
    FirebaseAnalytics.prototype.setEnabled = function (enabled) { return cordova(this, "setEnabled", { "sync": true }, arguments); };
    FirebaseAnalytics.prototype.setCurrentScreen = function (name) { return cordova(this, "setCurrentScreen", { "sync": true }, arguments); };
    FirebaseAnalytics.prototype.resetAnalyticsData = function () { return cordova(this, "resetAnalyticsData", { "sync": true }, arguments); };
    FirebaseAnalytics.pluginName = "FirebaseAnalytics";
    FirebaseAnalytics.plugin = "cordova-plugin-firebase-analytics";
    FirebaseAnalytics.pluginRef = "cordova.plugins.firebase.analytics";
    FirebaseAnalytics.repo = "https://github.com/chemerisuk/cordova-plugin-firebase-analytics";
    FirebaseAnalytics.platforms = ["Android", "iOS"];
FirebaseAnalytics.ɵfac = function FirebaseAnalytics_Factory(t) { return ɵFirebaseAnalytics_BaseFactory(t || FirebaseAnalytics); };
FirebaseAnalytics.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: FirebaseAnalytics, factory: function (t) { return FirebaseAnalytics.ɵfac(t); } });
var ɵFirebaseAnalytics_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(FirebaseAnalytics);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FirebaseAnalytics, [{
        type: Injectable
    }], null, null); })();
    return FirebaseAnalytics;
}(IonicNativePlugin));
export { FirebaseAnalytics };
